go-controller/pkg/node/healthcheck.go

```diff
@@ -92,18 +122,29 @@ func (l *loadBalancerHealthChecker) DeleteEndpointSlice(epSlice *discovery.Endpo
 	namespacedName := namespacedNameFromEPSlice(epSlice)
 	l.Lock()
 	defer l.Unlock()
-	delete(l.endpoints, namespacedName)
+	epSlices, err := l.watchFactory.GetEndpointSlices(epSlice.Namespace, epSlice.Labels[discovery.LabelServiceName])
```

Comment:
@tssurya kinda feels like this should be a function itself since we use this pattern in 3 different places.

If we fail to get endpoint slices in the add/update case, should we delete the namespacedName from the list, and wait for the next update? (eg, like delete does)

Setting the endpoint count to zero, and deleting from l.endpoints have the same effect; the healthcheck code sees `hcs.services[nsn].endpoints = 0` in both cases.

Reply:
yes true, I can remove the code redundancy. let me clean this up.

Reply:
sorry missed the second part of your comment, yes I took the easy route like you said to just set it to zero :D

Reply:
oh making it like delete will help me converge the code better. gotcha

Reply:
Changed to the logic Dan said as it makes this cleaner,
```
	for nsn, hci := range hcs.services {
		if _, found := newEndpoints[nsn]; !found {
			hci.endpoints = 0
		}
	}
```
will take care of nil key in cache case.